components/vfs/archive.hpp

```diff
@@ -21,6 +21,9 @@ namespace VFS
     public:
         virtual ~Archive() {}
 
+        /// Clears cached data for archives that may change.
```

Comment:
Is there any reason to treat BSAs specially? How do we know they don't change? Anyway, the VFS class isn't supposed to be concerned with details of the underlying archives.

Reopen the BSA should be pretty cheap, all that happens is parsing the header.